features/netsocket/NetworkInterface.h

```diff
@@ -352,13 +352,15 @@ class NetworkInterface: public DNS {
      */
     void add_event_listener(mbed::Callback<void(nsapi_event_t, intptr_t)> status_cb);
 
+#if MBED_CONF_PLATFORM_CALLBACK_COMPARABLE
     /** Remove event listener from interface.
      *
      * Remove previously added callback from the handler list.
      *
      *  @param status_cb The callback to unregister.
      */
     void remove_event_listener(mbed::Callback<void(nsapi_event_t, intptr_t)> status_cb);
```

Comment:
Would be good to mention in doxygen that this function is not available if the callback_comparable flag is false, it may be not obvious if user looks at the doxygen and handbook only.